frontend/templates/product-list/hooks/useVariantProductList.ts

```diff
@@ -0,0 +1,43 @@
+import { ProductList } from '@models/product-list';
+import {
+   destylizeDeviceTitle,
+   destylizeDeviceTitleAndVariant,
+   splitDeviceAndVariant,
+} from '@helpers/product-list-helpers';
+
+export function useVariantProductList(
+   productList: ProductList,
+   deviceHandle?: string
+) {
+   if (!deviceHandle) return productList;
+   const { variant } = splitDeviceAndVariant(deviceHandle);
+   if (!variant) return productList;
+
+   const modelNumberPattern = productList.wikiInfo?.find(
+      (info) => info.name === 'model_number_pattern'
+   )?.value;
+   const modelNumberRegex = modelNumberPattern
+      ? new RegExp(modelNumberPattern?.replace(/[*⎵]/g, '.'), 'g')
```

Comment:
I think using `replaceAll(modelNumberPattern, variant)` would be simpler than constructing a global regex to pass to replace.